File: src/main/java/com/project/webapp/sportmanager/model/Training.java
package com.project.webapp.sportmanager.model;

import com.project.webapp.sportmanager.utils.SessionType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "trainings")
public class Training {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private LocalDateTime startTime;

    @NotNull
    private LocalDateTime endTime;

    @NotNull
    @Enumerated(EnumType.STRING)
    private SessionType sessionType;

    @ManyToOne
    private Athlete athlete;

    @Column(length = 500, nullable = false)
    private String description;

    private Boolean athletePresence;
    private Boolean completing;

    @Column(length = 200)
    private String athleteFeelings;

}

File: frontend/src/app/model/training.ts
import {SessionType} from "../utils/session-type";
import {Athlete} from "./athlete.model";

export class Training {

  id: number | null;
  startTime: string | null;
  endTime: string | null;
  athlete: Athlete | null;
  sessionType: SessionType | null;
  description: string | null;
  athletePresence: boolean | null;
  completing: boolean | null;
  athleteFeelings: string | null;

  constructor(id: number | null, startTime: string | null, endTime: string | null, athlete: Athlete | null,
              sessionType: SessionType | null, description: string | null,
              athletePresence: boolean | null, completing: boolean | null, athleteFeelings: string | null) {
    this.id = id;
    this.startTime = startTime;
    this.endTime = endTime;
    this.athlete = athlete;
    this.sessionType = sessionType;
    this.description = description;
    this.athletePresence = athletePresence;
    this.completing = completing;
    this.athleteFeelings = athleteFeelings;
  }

  public static register(startTime: string | null, endTime: string | null,
                         sessionType: SessionType | null, description: string | null) {
    return new Training(null, startTime, endTime, null, sessionType, description,
      null, null, null);
  }

}
